core/src/avm2/activation.rs

```diff
@@ -1086,8 +1086,8 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         let function: Result<Object<'gc>, Error> = receiver
             .get_method(index.0)
             .ok_or_else(|| format!("Object method {} does not exist", index.0).into());
-        let base_proto = receiver.proto();
-        let value = function?.call(Some(receiver), &args, self, base_proto)?;
+        let superclass_object = receiver.instance_of();
```

Comment:
General note: the naming over here is confusing. We get an object's class with `instance_of`, store it in variable `superclass_object` (which in our naming is an entirely different thing), which is then passed into argument and Activation field named `subclass_object`.

Reply:
Right. So, `superclass_object` is supposed to represent the class that defined a particular method, so that subsequent calls to `super` progress up the class chain correctly. `super` is also supposed to work on bound methods, which is why we currently pull `instance_of` in a lot of places where we have a bare method. I suspect the correct thing to do would be to actually eliminate the `superclass_object` parameter from `call` entirely and just have methods bind to the class they were defined on (in which case we could rename this to `defining_class`).

However, I'm not going to do that until I implement lazy method binding, as I suspect that feature will interact with bound defining classes in an unintuitive way.